### PR TITLE
Improve `update_icons.dart` script

### PR DESCRIPTION
**Pull request by guidezpl**
#### Safety
- Adds a safety check for codepoint stability (from now on, existing codepoints won't change between updates)
#### Options
- Adds an option to enable/disable the safety checks
- Adds an option to regenerate `icons.dart` from a specified file
- Adds an option to specify the `fontFamily`
#### Other
- Minor improvements in naming and comments
- Fixes dartdoc for `insert_chart`'s  outlined variant

### Prepare for upcoming update
- Adjusts the identifier rewrites 
- Remove brittle and special handling for certain icons
- Handle incoming unstyled codepoints, they now come suffixed with `_baseline`

### Tests
- Adds tests for safety checks
- Validates that running this script on the existing codepoints produces an identical `icons.dart`

### CODE REVIEW

File: dev/tools/update_icons.dart [hunk 1]
```diff
@@ -185,33 +196,51 @@ void main(List<String> args) {
...
   final Map<String, String> oldTokenPairMap = _stringToTokenPairMap(oldCodepointsString);
 
-  _testIsMapSuperset(newTokenPairMap, oldTokenPairMap);
-
-  final String iconClassFileData = iconClassFile.readAsStringSync();
+  stderr.writeln('Performing safety checks');
+  final bool isSuperset = testIsSuperset(newTokenPairMap, oldTokenPairMap);
+  final bool isStable = testIsStable(newTokenPairMap, oldTokenPairMap);
+  if ((!isSuperset || !isStable) && argResults[_enforceSafetyChecks] as bool) {
+    exit(1);
```
Comment (rami-a) [inline dev/tools/update_icons.dart:203]: Nit: Should this print to stderr here before exiting

File: dev/tools/update_icons.dart [hunk 2]
```diff
@@ -349,14 +408,15 @@ class _Icon {
       htmlSuffix = '-filled';
     } else {
       family = 'material';
-      if (id.endsWith('_outlined') && id != 'insert_chart_outlined') {
```
Comment (rami-a) [inline dev/tools/update_icons.dart:352]: Is checking for `insert_chart_outlined` no longer necessary?

File: packages/flutter/lib/src/material/icons.dart [hunk 3]
```diff
@@ -10287,7 +10286,7 @@ class Icons {
   /// <i class="material-icons-round md-36">insert_chart</i> &#x2014; material icon named "insert chart" (round).
   static const IconData insert_chart_rounded = IconData(0xf819, fontFamily: 'MaterialIcons');
 
-  /// <i class="material-icons md-36">insert_chart_outlined</i> &#x2014; material icon named "insert chart outlined".
+  /// <i class="material-icons-outlined md-36">insert_chart</i> &#x2014; material icon named "insert chart" (outlined).
```
Comment (rami-a) [inline packages/flutter/lib/src/material/icons.dart:10289]: Ah I see, so we changed that one that I mentioned above?

**Review by rami-a — APPROVED** [review]
LGTM

File: dev/tools/update_icons.dart (hunk 2 above)
Comment (guidezpl) [reply dev/tools/update_icons.dart:352]: Exactly

File: packages/flutter/lib/src/material/icons.dart (hunk 3 above)
Comment (guidezpl) [reply packages/flutter/lib/src/material/icons.dart:10289]: Yeah I double checked with the codepoints, it should've been the outlined variant from the start

**Comment by guidezpl** [comment]
Merging as test failure is unrelated

**Pull request merged**